File: Elixir/zoo.exs
defmodule Animal do
    def info({:dog, name}) do
        "#{name} is a dog."
    end
    def info({:cat, name}) do
        "#{name} is a cat."
    end
end

zoo = [
    {:dog, "Charlie"}, {:dog, "Milo"},
    {:cat, "Gracie"}, {:cat, "Molly"}
]

for x <- zoo do
    IO.puts Animal.info(x)
end

File: Go/zoo.go
package main

import "fmt"

type Animal interface {
	info() string
}

type Cat struct {
	name  string
}

type Dog struct {
	name string
}


func (c Cat) info() string {
	return c.name + " is a cat."
}

func (d Dog) info() string {
	return d.name + " is a dog."
}


func main() {
	zoo := []Animal{Dog{name: "Charlie"}, Cat{name: "Molly"}}

	for _, x := range zoo {
		fmt.Println(x.info())
	}
}
